model/autocomplete.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package model
+
+import (
+	"encoding/json"
+	"io"
+)
+
+type UserAutocomplete struct {
+	In  []*User `json:"in"`
```

Comment:
Yeah, I know what you mean about it being confusing. Really what I wanted it to mean was in/out of the context. The context could be the channel or the team, and both contexts can be queried through the same endpoint so I wanted it contained within a single struct so the client doesn't need to parse different structs based on different query parameters it provides.

I knew it would be confusing so I specifically mentioned it in the API ref and the driver comment.

Any ideas on how to keep it a single struct and make it more clear?